core/src/main/kotlin/com/alecstrong/sql/psi/core/SqlCoreEnvironment.kt

```diff
@@ -203,6 +222,8 @@ private class CoreFileIndex(
       val file = localFileSystem.findFileByPath(it.absolutePath)
         ?: throw NullPointerException("File ${it.absolutePath} not found")
       iterateContentUnderDirectory(file, iterator)
+    } && predefinedTables.value.all {
+      iterator.processFile(it)
```

Comment:
does this cause these predefined tables to appear in the generated code? IE in the onCreate method

Reply:
Yes! And it is expected, otherwise, you can't return its values: `SELECT * FROM dual`, the dummy table used by Oracle. But I definitely only want to generate the file, if it is really used by sqldelight/the queries to don't blow up your classes.
And I have to test it, if shading the package name is needed. My prototype worked with `dependency(projects.base)`, but I will test it further. But this is the part of sqldelight :)

Reply:
if these are system level tables they shouldn't appear codegend (also tbc I'm referring to it appearing in [this code](https://github.com/cashapp/sqldelight/blob/master/sqldelight-compiler/integration-tests/src/test/kotlin/com/example/testmodule/TestDatabaseImpl.kt#L41-L72))

Reply:
Do I miss something? What do you mean? 😅
Hm, if we skip generating them, how do you query them? You would be forced to add a manual mapper. And for 'INSERT' too. We could make them internal. 
And like I said before, I really only want to generate them lazily.

Reply:
so it should still be possible to query these tables, ie writing 

```
myQuery:
SELECT *
FROM SYSIBM.SYSDUMMY1
```

does generate the query code for calling this, but if I call `MyDatabase.Schema.onCreate()` it doesn't try to `CREATE TABLE SYSIBM.SYSDUMMY1`

Reply:
Yeah, of course, it is not included in the create table statements, otherwise, it will crash. I implemented it in my db2 dialect and it works. 
![Bildschirm­foto 2022-10-01 um 17 32 37 (2)](https://user-images.githubusercontent.com/22521688/193416949-30907666-eef6-445a-b495-286d5c0922fd.png)